lib/elixir/lib/map_set.ex

```diff
@@ -63,6 +63,61 @@ defmodule MapSet do
   # with Elixir 1.14 and earlier and executed on Elixir 1.15+.
   defstruct map: :sets.new(version: 2)
 
+  @doc """
+  Checks if `map_set` contains `value`. Allowed in guards.
+
+  Otherwise functions like `member?/2`.
+
+  ## Examples
+
+      iex> MapSet.is_member(MapSet.new([1, 2, 3]), 2)
+      true
+
+      iex> MapSet.is_member(MapSet.new([1, 2, 3]), 4)
+      false
+
+      iex> case MapSet.new([1, 2, 3]) do
+      ...>   map_set when MapSet.is_member(map_set, 3) -> true
+      ...>   _ -> false
+      ...> end
+      true
+
+  ## Exceptions
+
+  Normally, if `map_set` is not a `MapSet`, `is_member/2` will raise (just like `member?/2`).
+  However, in guards, it will simply return false, allowing chaining.
+
+      iex> MapSet.is_member([1, 2, 3], :value)
+      ** (ArgumentError) expected a MapSet, got: [1, 2, 3]
+
+      iex> case [1, 2, 3] do
+      ...>   enumerable when MapSet.is_member(enumerable, 3) or length(enumerable) == 3 -> true
+      ...>   _ -> false
+      ...> end
+      true
+
+  """
+  @doc guard: true, since: "1.17.0"
+  defmacro is_member(map_set, value) do
+    case Macro.Env.in_guard?(__CALLER__) do
+      true ->
+        quote do
+          is_struct(unquote(map_set), unquote(__MODULE__)) and
+            is_map_key(unquote(map_set).map, unquote(value))
```

Comment:
Sorry, there was a misunderstanding. `is_map_key` raises in guards. `is_member` should also raise in guards.

```suggestion
          (unquote(map_set).__struct__ == unquote(__MODULE__) or :fail) and
            is_map_key(unquote(map_set).map, unquote(value))
```

Reply:
Ah, ok—that makes sense (being consistent in both forms).